app/frontend/src/features/profile/edit/EditHostingPreference.test.tsx

```diff
@@ -1,42 +1,54 @@
 import { render, screen } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
 import { Route, Switch } from "react-router-dom";
-import { userRoute } from "routes";
+import { editHostingPreferenceRoute, userRoute } from "routes";
 import { service } from "service";
-import wrapper from "test/hookWrapper";
+import { getHookWrapperWithClient } from "test/hookWrapper";
 import { getUser } from "test/serviceMockDefaults";
 
 import { addDefaultUser, MockedService } from "../../../test/utils";
 import { SAVE } from "../../constants";
 import EditHostingPreference from "./EditHostingPreference";
 
-const mockProfilePage = "Mock Profile Page";
 const getUserMock = service.user.getUser as MockedService<
   typeof service.user.getUser
 >;
-const MockProfileComponent = () => <div title={mockProfilePage} />;
+const updateHostingPreferenceMock = service.user
+  .updateHostingPreference as MockedService<
+  typeof service.user.updateHostingPreference
+>;
+
+const renderPage = () => {
+  const { wrapper } = getHookWrapperWithClient({
+    initialRouterEntries: [`${editHostingPreferenceRoute}`],
+  });
+
+  render(
+    <Switch>
+      <Route path={editHostingPreferenceRoute}>
+        <EditHostingPreference />
+      </Route>
+      <Route path={userRoute}>
+        <h1 data-testid="user-profile">Mock Profile Page</h1>
+      </Route>
+    </Switch>,
+    { wrapper }
+  );
+};
 
 describe("EditHostingPreference", () => {
   beforeEach(() => {
-    getUserMock.mockImplementation(getUser);
     addDefaultUser();
+    getUserMock.mockImplementation(getUser);
+    updateHostingPreferenceMock.mockResolvedValue(new Empty());
   });
 
-  it.skip("should redirect to Profile route after successful update", async () => {
-    render(
-      <Switch>
-        <Route path={userRoute}>
-          <MockProfileComponent />
-        </Route>
-        <Route>
-          <EditHostingPreference />
-        </Route>
-      </Switch>,
-      { wrapper }
-    );
+  it("should redirect to Profile route after successful update", async () => {
+    renderPage();
 
     userEvent.click(await screen.findByRole("button", { name: SAVE }));
 
-    expect(await screen.findByTitle(mockProfilePage)).toBeInTheDocument();
-  });
+    expect(await screen.findByTestId("user-profile")).toBeInTheDocument();
+  }, 20000);
```

Comment:
So it looks like it's not to do with the markdown editor. I wonder what makes this take that much longer to run than other tests that this needs an increased timeout value...

Reply:
Maybe it's just a heavier page, frontend tests take quite a long time in CI

Reply:
Yeah the markdown editor must be more heavy to render...